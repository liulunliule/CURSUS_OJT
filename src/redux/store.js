import { combineReducers, configureStore } from "@reduxjs/toolkit";
import savedCourseReducer from "./features/savedCourseSlice";
import blogReducer from './features/ourBlogSlice';
import officesReducer from './features/officesSlice';
import pressReducer from './features/pressSlice';
import payoutReducer from './features/payoutSlice';
import creditsReducer from "./features/creditsSlice";
import certificateReducer from "./features/myCertificateSlice";
import myProfileReducer from "./features/myProfileSlice";
import myHeaderReducer from "./features/myHeaderSlice";
import mySearchReducer from "./features/mySearchSlice";
import myStudioReducer from "./features/myStudioSlice";
import allInstructorReducer from "./features/allInstructorSlice";
import reviewFeedbackReducer from "./features/reviewfeedbackSlice";
import feedbackReportReducer from "./features/feedbackvsReportSlice";
import notificationsSettingReducer from "./features/NotificationsSettingSlice";
import privacySettingReducer from "./features/PrivacySettingSlice";
import userSlice from "./features/userSlice";
import certificationTestPageReducer from "./features/certificationTestPageSlice";
import certificationTestResultReducer from "./features/certificationTestResultSlice";
import analyticsReducer from "./features/analyticsSlice";
import storage from "redux-persist/lib/storage";
import { persistReducer, persistStore } from "redux-persist";
import courseSlice from "./features/courseSlice";
import checkoutReducer from './features/checkoutSlice';
import earningReducer from './features/earningSlice';
import statementReducer from './features/statementSlice';
import shoppingCartReducer from './features/shoppingCartSlice';

const persistConfig = {
    key: "root",
    storage,
    whitelist: [
        "user",
        "savedCourse",
        "certificate",
        "course",
        "myProfile",
        "myHeader",
        "mySearch",
        "myStudio",
        "allInstructor",
        "reviewFeedback",
        "feedbackReport",
        "notificationsSetting",
        "privacySetting",
    ],
};

const rootReducer = combineReducers({
    user: userSlice,
    savedCourse: savedCourseReducer,
    certificate: certificateReducer,
    certificationTestPage: certificationTestPageReducer,
    certificationTestResult: certificationTestResultReducer,
    analytics: analyticsReducer,
    course: courseSlice,
    myProfile: myProfileReducer,
    myHeader: myHeaderReducer,
    mySearch: mySearchReducer,
    myStudio: myStudioReducer,
    allInstructor: allInstructorReducer,
    reviewFeedback: reviewFeedbackReducer,
    feedbackReport: feedbackReportReducer,
    checkout: checkoutReducer,
    earnings: earningReducer,
    statement: statementReducer,
    credits: creditsReducer,
    payout: payoutReducer,
    blogs: blogReducer,
    offices: officesReducer,
    press: pressReducer,
    notificationsSetting: notificationsSettingReducer,
    privacySetting: privacySettingReducer,
    shoppingCart: shoppingCartReducer,
});

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = configureStore({
    reducer: persistedReducer,
});

export const persistor = persistStore(store);
