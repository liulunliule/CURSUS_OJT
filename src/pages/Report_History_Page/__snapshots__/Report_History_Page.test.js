import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { store } from '../../../redux/store';
import ReportHistoryPage from '..';

test('test Certification Test Page', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <ReportHistoryPage/>
      </MemoryRouter>
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
