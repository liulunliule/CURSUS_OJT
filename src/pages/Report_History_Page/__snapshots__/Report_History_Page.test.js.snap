// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ReportHistoryPage 1`] = `
<div
  className="ReportHistoryPage "
>
  <div
    className="wrapper_ReportHistoryPage__Title"
  >
    <img
      alt=""
      className="ReportHistoryLogo_ReportHistoryPage"
      src="ReportHistoryLogo_SideBar.png"
    />
    <h2
      className="ReportHistoryPage__Title"
    >
      Report history
    </h2>
  </div>
  <div
    className="wrapper_ReportHistoryPage__TitleSmall"
  >
    <h4
      className="ReportHistoryPage__TitleSmall"
    >
      Thanks for reporting
    </h4>
  </div>
  <div
    className="wrapper_ReportHistoryPage__content"
  >
    <p>
      Any member of the Cursus community can flag content to us that they believe violates our Community Guidelines. When something is flagged, itâ€™s not automatically taken down. Flagged content is reviewed in line with the following guidelines:
    </p>
    <ul>
      <li>
        Content that violates our
         
        <a
          className="Link_ReportHistoryPage"
          href="/"
          onClick={[Function]}
        >
          Community Guidelines
        </a>
         
        is removed from Edututs+.
      </li>
      <li>
        Content that may not be appropriate for all younger audiences may be age-restricted.
      </li>
    </ul>
    <a
      className="Link_ReportHistoryPage"
      href="/"
      onClick={[Function]}
    >
      <p>
        Learn more about reporting content on Cursus.
      </p>
    </a>
  </div>
  <div
    className="wrapper_ReportHistoryPage__end"
  >
    <span>
      You haven't submitted any reports.
    </span>
  </div>
</div>
`;
